Add developer mailing list to fledge/index.md (#5301)
GoogleChrome/developer.chrome.com: site/en/docs/privacy-sandbox/fledge/index.md

-  **GitHub**: Read the [proposal](https://github.com/WICG/turtledove/blob/master/FLEDGE.md),
   [raise questions and follow discussion](https://github.com/WICG/turtledove/issues).

> -  **FLEDGE API announcements**: Join or view the mailing list at: https://groups.google.com/u/0/a/chromium.org/g/fledge-api-announce.

Reviewer: ```suggestion
-  **Announcements**: Join or view past announcements on the [FLEDGE mailing list](https://groups.google.com/u/0/a/chromium.org/g/fledge-api-announce).
```
